components/TextField.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useState, useEffect, useRef, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  Input,
+  InputProps as IProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+interface TextFieldProps extends FlexProps {
+  label: string
+  value?: string
+  InputProps?: IProps
+  LeftAddons?: ReactNode
+  RightAddons?: ReactNode
+}
+
+const TextField: FC<TextFieldProps> = ({
+  label,
+  value = '',
+  InputProps = {},
+  LeftAddons = null,
+  RightAddons = null,
+  ...props
+}) => {
+  const [val, setVal] = useState<string>(value)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const styles = useMultiStyleConfig('TextField', props)
+
+  useEffect(() => {
+    setVal(value)
+  }, [value])
+
+  return (
+    <Flex
+      id="TextField"
+      tabIndex={0}
+      onFocus={() => inputRef?.current?.focus()}
+      {...props}
+      sx={{
+        ...styles?.container,
```

Comment:
Unfamiliar with this offhand -- Am I correct in thinking that the properties here come from [theme.ts](https://github.com/iron-fish/ui-kit/pull/7/files#diff-6ecb222b58bd55a2f5fab1aaccba9193a3f4e92309a433c1e327cce844020834R51) ?

Reply:
Yes, you are correct.